src/table.rs

```diff
@@ -94,14 +98,29 @@ pub fn expand_table(input: &mut DeriveInput) -> Result<TokenStream> {
 }
 
 fn get_table_meta(id: &Ident, attrs: &[Attribute]) -> Result<TableMeta> {
-    let mut table_name = LitStr::new(&id.to_string(), id.span());
+    let mut table_name = None;
     let mut case = Case::default();
 
     for attr in attrs {
         if attr.path().is_ident(AWS_DYNAMO_ATTR_META_ENTRY) {
             attr.parse_nested_meta(|table| {
                 if table.path.is_ident(KEY_TABLE_NAME) {
-                    table_name = table.value()?.parse()?;
+                    if table_name.is_some() {
+                        return Err(Error::new(
+                            table.path.span(),
+                            "table name already set. static str and function cannot be used both",
+                        ));
+                    }
+                    table_name = Some(TableName::StaticStr(table.value()?.parse()?));
+                }
+                if table.path.is_ident(KEY_TABLE_NAME_FN) {
+                    if table_name.is_some() {
+                        return Err(Error::new(
+                            table.path.span(),
+                            "table name already set. static str and function cannot be used both",
+                        ));
+                    }
+                    table_name = Some(TableName::Function(table.value()?.parse()?));
```

Comment:
```suggestion
                match table.path.get_ident() {
                    Some(id) if id == KEY_TABLE_NAME => {}
                    Some(id) if id == KEY_TABLE_NAME_FN => {}
                    Some(id) if id == KEY_RENAME => {}
                    _ => {}
                };
```
각 if block이 disjoint 한 것 같은데 이런 느낌의 패턴 매칭은 어떨까요?

Reply:
None일 시 empty arm이 조금 마음에 걸려서 이대로 가보고 가독성이 필요하면 도입하도록 하겠습니다 ㅠ.ㅠ